### PR TITLE
update to work with latest GHC

### PR DESCRIPTION
**Pull request by vmchale**
### Description

This just makes every monoid instance into a semigroup instance. 

### Checklist

  - [x] I've read [CONTRIBUTING.md](https://github.com/xmonad/xmonad/blob/master/CONTRIBUTING.md)

  - [x] I've confirmed these changes don't belong in xmonad-contrib instead

  - [x] I tested my changes with [xmonad-testing](https://github.com/xmonad/xmonad-testing)

  - [x] I updated the `CHANGES.md` file

### CODE REVIEW

**Comment by MichielDerhaeg** [comment]
If I recall correctly, the lowest version of GHC that we support is 7.6.3. The `Data.Semigroup` module was added with GHC 8.0. So I'm afraid you can't fix it this way.

**Comment by MichielDerhaeg** [comment]
Simply adding
```
if !impl(ghc >= 8.0)
  build-depends: semigroups == 0.18.*
```
might be the best solution. If everyone agrees to add another dependency.

**Comment by MichielDerhaeg** [comment]
Nevermind i tested it and it works with older versions like this.

**Comment by MichielDerhaeg** [comment]
@byorgey could you merge?

File: xmonad.cabal [hunk 1]
```diff
@@ -32,7 +32,8 @@ tested-with:
                     GHC==7.6.3,
                     GHC==7.8.4,
                     GHC==7.10.3,
-                    GHC==8.0.1
+                    GHC==8.0.1,
+                    GHC==8.2.2
```
Comment (byorgey) [inline xmonad.cabal:36]: Shouldn't we add `GHC==8.4.1` as well?

File: CHANGES.md [hunk 2]
```diff
@@ -18,6 +18,8 @@
     the focus follows when the mouse enters another workspace
     but not moving into any window.
 
+  * Compiles with GHC 8.4.0
```
Comment (byorgey) [inline CHANGES.md:21]: This should say `8.4.1` now that it has actually been released.

**Pull request merged**